res/css/views/rooms/_AppsDrawer.pcss

```diff
@@ -279,28 +271,58 @@ limitations under the License.
         animation-delay: 500ms;
         animation-name: mx_AppTileBody_fadeInSpinnerAnimation;
     }
-}
 
-.mx_AppTileBody {
-    --AppTileBody-height: 100%;
+    &.mx_AppTileBody--normal,
+    &.mx_AppTileBody--mini {
```

Comment:
Why are these extra selectors necessary? Is there a precedence issue we should be aware of?

Reply:
As noted on the description above, `mx_AppTileBody` now handles common declarations (such as [`border-radius: 8px`](https://github.com/matrix-org/matrix-react-sdk/pull/11002/files#diff-a94695056f674ec44233ef1dd8163588d10278cadaa3258d0a1112b32ec17d7eR260)) which must be applied to any variants of it. Combining both `.mx_AppTileBody--large` and `mx_AppTileBody--mini` with `mx_AppTileBody` here and not repeating those selectors somewhere else clarifies that the common declarations are applied to elements with the selectors, ensuring developers who are not yet familiar with the codebase or the naming policy that an unexpected regression should not happen as long as they add, edit, or remove declarations inside the nested style blocks.

It means I do not want this like the current develop branch, because it took for myself a fairly a lot of time to clearly understand the style cascading structure with confidence, especially regarding the relationship between the selectors, how they depend each other, whether or not the declaration specified later will override the one specified earlier, etc.:
````
.mx_AppTileBody {
    ...
}

.mx_AppTileBody--large,
.mx_AppTileBody--mini {
    ...
}

.mx_AppTileBody--large {
    ...
}

...
````

It adds nothing valuable to keep those selectors on the same level loosely. It not only exposes the elements to be styled with them to a chance of a unexpected regression in case the same selector, something like `div[class$='large']` is imported later than this file (on user's custom themes or not), or something we are not aware of, but also increases a chance of making scattered selectors orphan like [forever](https://github.com/matrix-org/matrix-react-sdk/pulls?q=is%3Apr+archaeology+is%3Aclosed) until someone cleans them up (which should be avoided in the first place) because [there is not (going to be) a linter to check unused selectors](https://github.com/vector-im/element-web/issues/25149).

Though I think that providing the exact specificity to elements to be styled is one of the most stable, sustainable, and practical ways currently available for us of ensuring code-wise that the elements are styled as expected, since we cannot depend completely on Percy due to a limited budget, making it possible for UI developers to add, edit, or remove declarations with confidence at the same time, I am open to a better solution if any, which can be placed.

Reply:
If your question is about the reason why the declarations with those selectors are required here, the answer to it is that they was moved from here https://github.com/matrix-org/matrix-react-sdk/pull/11002/files#diff-a94695056f674ec44233ef1dd8163588d10278cadaa3258d0a1112b32ec17d7eL259-L268, so that they are not applied to `mx_AppTileBody--loading`, which is styled differently.